Error with slurm-23-11-0-1: variable has incomplete type 'hostlist_t'

Compiling MPICH 4.1.2 against slurm 23-11-0-1 fails with the following error:
```
libtool: compile:  cc -DHAVE_CONFIG_H -I. -I./include -I/usr/ports/net/mpich/work/mpich-4.1.2/src/pmi/src -I/usr/ports/net/mpich/work/mpich-4.1.2/src/mpl/include -I/usr/ports/net/mpich/work/mpich-4.1.2/src/mpl/include -I./lib -I./lib/tools/topo -I./lib/tools/bootstrap/include -I./lib/tools/bootstrap/utils -I./lib/tools/bootstrap/persist -I./lib/tools/demux -isystem /usr/local/include -I/usr/local/include -DNETMOD_INLINE=__netmod_inline_ofi__ -I/usr/local/include/json-c -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing -g -MT lib/tools/bootstrap/external/ll_launch.lo -MD -MP -MF lib/tools/bootstrap/external/.deps/ll_launch.Tpo -c lib/tools/bootstrap/external/ll_launch.c  -fPIC -DPIC -o lib/tools/bootstrap/external/.libs/ll_launch.o
--- lib/tools/bootstrap/external/slurm_query_node_list.lo ---
lib/tools/bootstrap/external/slurm_query_node_list.c:29:16: error: variable has incomplete type 'hostlist_t' (aka 'struct hostlist')
    hostlist_t hostlist;
               ^
/usr/local/include/slurm/slurm.h:1221:16: note: forward declaration of 'struct hostlist'
typedef struct hostlist hostlist_t;
               ^
1 error generated.

```

--- a/src/pm/hydra/lib/tools/bootstrap/external/slurm_query_node_list.c
+++ b/src/pm/hydra/lib/tools/bootstrap/external/slurm_query_node_list.c
@@ -26,7 +26,7 @@ static struct HYD_node *global_node_list = NULL;
 #if defined(HAVE_SLURM)
 static HYD_status list_to_nodes(char *str)
 {
-    hostlist_t hostlist;
+    hostlist_t *hostlist;
     char *host;
     int k = 0;
     HYD_status status = HYD_SUCCESS;

